src/tools/pub/package-layout.md

```diff
@@ -476,4 +473,33 @@ documentation generators, or other bits of automation.
 Unlike the scripts in `bin`, these are *not* for external users of the package.
 If you have any of these, place them in a directory called `tool`.
 
+## Project specific caching for tools
+
+{{site.alert.info}}
+  Do not check the `.dart_tool/` directory into source control.
+  Instead keep `.dart_tool/` in `.gitignore`.
+{{site.alert.end}}
+
+The `.dart_tool/` is created when you've run `dart pub get`.
+The directory used by various tooling for caching files specific to your project and/or local machine.
```

Comment:
... directory is used by various tools ...

"Various" suggests a number of things, but "tooling" is not countable.